### PR TITLE
Feature/Update validation for columnar

### PR DESCRIPTION
**Pull request by figueroa1395**
Update validation functions to support columnar data:

- [x] Update `validate_input_data` function.
- [x] Update `validate_batch_data` function.
- [x] Update tests:
  - [x] `test_input_validation.py`
  - [x] `test_batch_validation.py`

### CODE REVIEW

**Comment by figueroa1395** [comment]
At this moment, the columnar based batch data is validated by converting the full batch to row based, which may have a big memory overhead. However, once #699 is finished, this will be upgraded to do the conversion per individual batches.

File: src/power_grid_model/_utils.py
Comment (mgovers) [inline src/power_grid_model/_utils.py:313]:
```suggestion
    Note: Copies are made on a per-component basis; if a component is row based in both the input and the requested
```

File: src/power_grid_model/validation/validation.py
Comment (mgovers) [inline src/power_grid_model/validation/validation.py:139]:
```suggestion
    # TODO(figueroa1395): transform to columnar per single batch scenario once the columnar dataset python extension is finished
```

this is not strictly necessary; just a speed-up

File: tests/unit/validation/test_batch_validation.py
Comment (mgovers) [inline tests/unit/validation/test_batch_validation.py:59]: it's a nitpick, but `...` looks weird here. Maybe it's a good idea to be explicit.

```suggestion
    return compatibility_convert_row_columnar_dataset(original_batch_data, Ellipsis, DatasetType.update)
```

it's equivalent, but more clearly tells the reader that it's a deliberate special value that is different than just "everything"/"et cetera" (which is what the ellipsis stands for)

the way i read the current state is also:
* "return (for compatibility) a converted row/columnar data set from the original batch data etc. as an update data set type".

as opposed to
* "return (for compatibility) a converted row/columnar data set from the original batch, using ellipsis and update data set type as parameters"

alternatively, you can also use explicit keyword arguments

File: tests/unit/validation/test_input_validation.py
Comment (mgovers) [inline tests/unit/validation/test_input_validation.py:280]: idem

File: src/power_grid_model/validation/validation.py
Comment (figueroa1395) [reply src/power_grid_model/validation/validation.py:139]: Yup, but since everything validation related is done on the Python side, we may as well try to do it as "fast" as possible. Hence, even though this is probably mergable, I will wait a bit since there is no rush.

**Review by mgovers — APPROVED** [review]
since validation for columnar data is not blocked by the Python API (since we already have preliminary python support), it's also OK to merge this. your call

**Comment by figueroa1395** [comment]
Merging now and doing the incremental upgrade to per-batch scenario validation later. This way the functionality is already provided to the users.

**Pull request merged**
